### PR TITLE
feat: added metric page in messaging queues

### PR DESCRIPTION
**Pull request by SagarRajput-7**
### Summary

Added metric page with 5 categories and 22 dashboards 

#### Screenshots

Overview Page - 
image

https://github.com/user-attachments/assets/87dbc055-c0b7-4003-a38a-65db1eca267f

#### Affected Areas and Manually Tested Areas

- manually tested and compared the dashboard with the given Kafka overview dashboard

----

> [!IMPORTANT]
> Added a new metric page for Kafka in messaging queues with detailed graphs and metrics.
> 
>   - **Behavior**:
>     - Added `MetricPage` component in `MQDetailPage.tsx` for Kafka metrics.
>     - New view type `metricPage` in `MessagingQueuesViewType` in `MessagingQueuesUtils.ts`.
>     - Updated `MessagingQueues.tsx` to include navigation to the metric page.
>   - **Components**:
>     - Added `MetricPage`, `MetricColumnGraphs`, `MetricPageGridGraph` in `MQDetails/MetricPage/`.
>     - `MetricPage` has collapsible sections for metric categories.
>   - **Styles**:
>     - Added `MetricPage.styles.scss` for new metric page styling.
>   - **Utilities**:
>     - Added `MetricPageUtil.ts` for widget data for Kafka metrics.
>   - **Localization**:
>     - Added `messagingQueues.json` and `messagingQueuesKafkaOverview.json` for English and British English locales.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for 9a27917484b7ef58ee731f885e80b297d52bf6a1. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: frontend/src/pages/MessagingQueues/MessagingQueuesUtils.ts [hunk 1]
```diff
@@ -242,6 +242,10 @@ export const MessagingQueuesViewType = {
 		label: 'Drop Rate view',
 		value: 'dropRate',
 	},
+	metricPage: {
+		label: 'Metric Page',
```
[the remark this answers is not in the document]
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MessagingQueuesUtils.ts]: done

**Comment by SagarRajput-7** [comment]
Added doc not required as this is a part of a feature and can't have doc required for every single part PR, also we are making sure to have docs for Kafka 2.0

**Review by rahulkeswani101 — DISMISSED** [review]

**Review by rahulkeswani101 — APPROVED** [review]

**Review by rahulkeswani101 — APPROVED** [review]

File: frontend/src/pages/MessagingQueues/MQDetailPage/MQDetailPage.tsx [hunk 2]
```diff
@@ -112,21 +117,24 @@ function MQDetailPage(): JSX.Element {
...
 			)}
-			{selectedView !== MessagingQueuesViewType.dropRate.value && (
-				<div className="messaging-queue-details">
-					<MessagingQueuesDetails
-						selectedView={selectedView}
-						producerLatencyOption={producerLatencyOption}
-					/>
-				</div>
-			)}
+			{selectedView !== MessagingQueuesViewType.dropRate.value &&
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MQDetailPage/MQDetailPage.tsx:129]: Move to constants

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+	title,
+	description,
+	graphCount,
+}: MetricSectionProps): JSX.Element {
+	const isDarkMode = useIsDarkMode();
+
+	return (
+		<div className="metric-column-graph">
+			<CardContainer className="row-card" isDarkMode={isDarkMode}>
+				<div className={cx('row-panel')}>
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx:41]: remove cx

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+		</div>
+	);
+}
+
+function MetricColumnGraphs(): JSX.Element {
+	const metricsData = [
+		{
+			title: 'Broker Metrics',
+			description:
+				'The Kafka Broker metrics here inform you of data loss/delay through unclean leader elections and network throughputs, as well as request fails through request purgatories and timeouts metrics',
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx:65]: use strings

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+				consumerFetchRateWidgetData,
+				messagesConsumedWidgetData,
+			],
+			id: 'consumer-metrics',
+		},
+	];
+
+	return (
+		<div className="metric-column-graph-container">
+			{metricsData.map((metric) => (
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx:104]: Add safety check

File: frontend/src/pages/MessagingQueues/MessagingQueues.tsx [hunk 6]
```diff
@@ -192,7 +192,7 @@ function MessagingQueues(): JSX.Element {
 							</Button>
 						</div>
 					</div>
-					<div className="overview-info-card">
+					<div className="overview-info-card middle-card">
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MessagingQueues.tsx:195]: `middle-card` is being used to add additional styling, you can use nth child and access element directly

File: frontend/src/pages/MessagingQueues/MessagingQueues.tsx [hunk 7]
```diff
@@ -192,7 +192,7 @@ function MessagingQueues(): JSX.Element {
...
 						<div>
```
Comment (YounixM) [inline frontend/src/pages/MessagingQueues/MessagingQueues.tsx:196]: Empty div ?

Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MessagingQueues.tsx:196]: using these to categorise the child elements - `[(1,2) ---- 3]`

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx (hunk 4 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx]: done

File: frontend/src/pages/MessagingQueues/MQDetailPage/MQDetailPage.tsx (hunk 2 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetailPage/MQDetailPage.tsx]: done

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx (hunk 3 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx]: done

File: frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx (hunk 5 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MQDetails/MetricPage/MetricColumnGraphs.tsx:103]: done

File: frontend/src/pages/MessagingQueues/MessagingQueues.tsx (hunk 6 above)
Comment (SagarRajput-7) [reply frontend/src/pages/MessagingQueues/MessagingQueues.tsx]: done

**Review by YounixM — APPROVED** [review]

**Pull request merged**